storage/src/main/java/org/apache/kafka/storage/internals/log/PendingExpandIsr.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.storage.internals.log;
+
+import org.apache.kafka.metadata.LeaderAndIsr;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public record PendingExpandIsr(int newInSyncReplicaId,
+                               LeaderAndIsr sentLeaderAndIsr,
+                               CommittedPartitionState lastCommittedState) implements PendingPartitionChange {
+
+    @Override
+    public CommittedPartitionState lastCommittedState() {
+        return lastCommittedState;
+    }
```

Comment:
Do we need this implementation?
IIUC, `record` has already provided the exact same implementation behind the scenes.